python/tvm/tir/ir_builder.py

```diff
@@ -62,9 +62,10 @@ class BufferVar(ObjectGeneric):
     IRBuilder.allocate
     """
 
-    def __init__(self, builder, buffer_var, content_type):
+    def __init__(self, builder, buffer_var, shape, content_type):
```

Comment:
I don't think this change is necessary, but i would wait for your response for my other comment. Maybe after that we can discuss here :)

Reply:
I have one query here. What happens if we don't have the shape info. In that case, how do you enforce the check in _linear_index ?

For example: We create a buffer pointer out of certain existing buffer, in that case you may not have necessarily the shape info, in that case, how it works ?

Reply:
This is change okay. Post clarification!

Reply:
You can use shape=None and the code will work fine assuming you always use a single index.

Reply:
Request to add small comment section to describe new way to use it.
As the new cases are 2, either with already linear index or actual index. It will help understand the new change better.